src/Microsoft.AspNet.Razor/Text/SourceLocation.cs

```diff
@@ -15,34 +15,21 @@ public struct SourceLocation : IEquatable<SourceLocation>, IComparable<SourceLoc
         public static readonly SourceLocation Undefined = CreateUndefined();
         public static readonly SourceLocation Zero = new SourceLocation(0, 0, 0);
 
-        private int _absoluteIndex;
-        private int _lineIndex;
-        private int _characterIndex;
-
         public SourceLocation(int absoluteIndex, int lineIndex, int characterIndex)
         {
-            _absoluteIndex = absoluteIndex;
-            _lineIndex = lineIndex;
-            _characterIndex = characterIndex;
+            AbsoluteIndex = absoluteIndex;
+            LineIndex = lineIndex;
+            CharacterIndex = characterIndex;
         }
 
-        public int AbsoluteIndex
-        {
-            get { return _absoluteIndex; }
-        }
+        public int AbsoluteIndex { get; set; }
```

Comment:
Talked in person, just going to comment these things to avoid overkill of a custom serializer for a single type.